### ISSUE TITLE
Timeout waiting to lock after upgrade to Gradle 4.0.2

### ISSUE THREAD

**Issue by felipecsl** [gradle/gradle#2605]
After upgrading from Gradle 4.0 to 4.0.2, we started seeing a lot of build errors like the one below:

```
> Could not open Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13).
   > Timeout waiting to lock Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13). It is currently in use by another Gradle instance.
     Owner PID: 12479
     Our PID: 11313
     Owner Operation: 
     Our operation: 
     Lock file: /tmp/.gradle/caches/4.0.2/zinc-0.3.13/zinc-0.3.13.lock
```

### Context
The symptom seems similar to #851, but we're not sharing gradle caches.
Local and remote build cache are both disabled.
The problem seems to happen only on CI.

Here is the stacktrace:
```
-----------
* What went wrong:
Execution failed for task ':projects:affinity:compileScala'.
> Could not open Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13).
   > Timeout waiting to lock Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13). It is currently in use by another Gradle instance.
     Owner PID: 12479
     Our PID: 11313
     Owner Operation: 
     Our operation: 
     Lock file: /tmp/.gradle/caches/4.0.2/zinc-0.3.13/zinc-0.3.13.lock
* Try:
Run with --info or --debug option to get more log output.
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':projects:affinity:compileScala'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter.execute(SkipCachedTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:60)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:242)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:95)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:235)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:224)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:121)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:77)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:102)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:96)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:612)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:567)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:96)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.gradle.cache.CacheOpenException: Could not open Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13).
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:59)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:30)
	at org.gradle.cache.internal.DefaultCacheFactory.doOpen(DefaultCacheFactory.java:89)
	at org.gradle.cache.internal.DefaultCacheFactory.open(DefaultCacheFactory.java:63)
	at org.gradle.cache.internal.DefaultCacheRepository$PersistentCacheBuilder.open(DefaultCacheRepository.java:123)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompilerFactory.createParallelSafeCompiler(ZincScalaCompilerFactory.java:57)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler$Compiler.execute(ZincScalaCompiler.java:64)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:52)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:38)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$WorkerCompileSpec.compile(AbstractDaemonCompiler.java:74)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerDaemonServer.execute(AbstractDaemonCompiler.java:82)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerDaemonServer.execute(AbstractDaemonCompiler.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	... 6 more
Caused by: org.gradle.cache.internal.LockTimeoutException: Timeout waiting to lock Zinc 0.3.13 compiler cache (/tmp/.gradle/caches/4.0.2/zinc-0.3.13). It is currently in use by another Gradle instance.
Owner PID: 12479
Our PID: 11313
Owner Operation: 
Our operation: 
Lock file: /tmp/.gradle/caches/4.0.2/zinc-0.3.13/zinc-0.3.13.lock
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.lock(DefaultFileLockManager.java:269)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.<init>(DefaultFileLockManager.java:138)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:86)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:73)
	at org.gradle.cache.internal.FixedExclusiveModeCrossProcessCacheAccess.open(FixedExclusiveModeCrossProcessCacheAccess.java:59)
	at org.gradle.cache.internal.DefaultCacheAccess.open(DefaultCacheAccess.java:133)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:57)
	... 31 more
```

### Expected Behavior
The build would complete successfully.

### Current Behavior
Build fails on Scala compilation"
`Execution failed for task ':common:scalalib210:compileScala'.`

### Steps to Reproduce (for bugs)
The problem seems to go away by itself after a re-run.

**Comment by bmuschko** [comment]
Thanks for reporting the issue. Any chance you can go back to 4.0.1 or 4.0 to see if you still observe the issue?

**Comment by felipecsl** [comment]
That would be a big operation for us since it only happens in CI, i'd have to rollback and potentially run the risk of breaking the build for hundreds of engineers. I'm afraid this is not an option, unfortunately.

**Comment by big-guy** [comment]
You may be able to reproduce this locally by making sure `.gradle/caches/4.0.2/zinc-0.3.13` (in GRADLE_USER_HOME) is empty before building and running with the same number of `--max-workers`.

**Comment by felipecsl** [comment]
Seeing this exact same problem again now when upgrading from 4.0.2 to 4.1

**Comment by jjohannes** [comment]
My assumption is this: 
Many processes fight and wait for the same lock. Then one of them times out because it is not _its turn_ for a longer time. The timeout currently does not take into account that the log owner might change.

#2734 changes this by resetting the timeout if the lock owner changes. It includes some other changes to improve the efficiency of communication between the processes that fight for the same lock. In particular in situations where processes are slowing down (e.g. due to memory pressure). These changes _might_ also improve the situation here.

@felipecsl An issue with your build _could_ potentially be slow compiler daemons due to memory pressure. What are your memory settings (`forkOptions`) for the scala compile tasks? Might be worth doing some profiling to check if they have enough memory.

**Comment by felipecsl** [comment]
@jjohannes doesnt seem like we change the `forkOptions` anywhere, except for one module:
```
scalaCompileOptions.forkOptions.jvmArgs = ['-XX:MaxPermSize=512m']
```

**Comment by big-guy** [comment]
@jjohannes Another thing we could consider doing here is to make the zinc compiler interface cache cross-versionable, so you would only pay the penalty of generating the zinc compiler once vs on every Gradle upgrade.  There's still an issue with if there's enough load, the zinc compiler will take a long time creating the interface jar.  Maybe the newer versions of Zinc will be better at that (#2158).

I also don't know why we open the zinc cache with an exclusive lock initially:
https://github.com/gradle/gradle/blob/998e9696e3c461d3f7495f6d054ca633998d9cfd/subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompilerFactory.java#L60

This could probably be `None`, but I don't think it'll help this particular problem.

**Comment by jjohannes** [comment]
@big-guy 
> Another thing we could consider doing here is to make the zinc compiler interface cache cross-versionable, so you would only pay the penalty of generating the zinc compiler once vs on every Gradle upgrade. 

We can do that, but we still should fix it in a way that it also does not timeout the first time it creates the inteface.jar.

>There's still an issue with if there's enough load, the zinc compiler will take a long time creating the interface jar. 

If we are sure that this is what happens, we should increase the timeout (only for this one lock, not in general for all locks). Because then it is valid that other processes need to wait. Wdyt?

> Maybe the newer versions of Zinc will be better at that (#2158).

We should do that. Yes.

> I also don't know why we open the zinc cache with an exclusive lock initially.
This could probably be None, but I don't think it'll help this particular problem.

I think this won't make a difference here. Both take an exclusive lock. The difference is if the lock is closed directly or kept open until someone else requests it. But since the lock is only used once at the beginning of compilation, I think the current strategy is fine.

**Comment by jjohannes** [comment]
The cache locking improvements (#2734) are now merged and will be in 4.2.

@felipecsl I don't know if you have a possibility to try this out, but the changes are in this nightly `4.2-20170825090016+0000`. Otherwise I would encourage you to try 4.2 RC1 ones it is out (approx. in two weeks).

Should the problem persists, we can investigate further.

**Comment by dsilvasc** [comment]
I just ran into this with Gradle 4.3-20170911235713 and zinc 0.3.15:

```
Execution failed for task ':some:module:compileScala'.
> Could not open Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15).
   > Timeout waiting to lock Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15). It is currently in use by another Gradle instance.
     Owner PID: 17847
     Our PID: 17845
     Owner Operation: 
     Our operation: 
     Lock file: /home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15/zinc-0.3.15.lock
```

**Comment by jjohannes** [comment]
Thanks for letting us know @dsilvasc. Is there a chance that you can find a stacktrace for this in the daemon logs?

Also, can you confirm that it is not this issue: https://github.com/gradle/gradle/issues/851 (sharing gradle home in multiple containers)

**Comment by dsilvasc** [comment]
Looks like our CI system runs without stack traces, but I'll work on setting up a job with that enabled.

https://github.com/gradle/gradle/issues/851 might be related.  It's possible that our gradle run forks (or that we kick off multiple gradle runs in parallel).  I'll find out more details about that one too.

**Comment by dsilvasc** [comment]
@jjohannes  Stack trace:

```
* What went wrong:
Execution failed for task ':some:project:compileScala'.
> Could not open Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15).
   > Timeout waiting to lock Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15). It is currently in use by another Gradle instance.
     Owner PID: 16501
     Our PID: 16540
     Owner Operation: 
     Our operation: 
     Lock file: /home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15/zinc-0.3.15.lock

* Try:
Run with --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':some:project:compileScala'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.gradle.cache.CacheOpenException: Could not open Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15).
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:61)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:32)
	at org.gradle.cache.internal.DefaultCacheFactory.doOpen(DefaultCacheFactory.java:90)
	at org.gradle.cache.internal.DefaultCacheFactory.open(DefaultCacheFactory.java:64)
	at org.gradle.cache.internal.DefaultCacheRepository$PersistentCacheBuilder.open(DefaultCacheRepository.java:123)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompilerFactory.createParallelSafeCompiler(ZincScalaCompilerFactory.java:58)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler$Compiler.execute(ZincScalaCompiler.java:64)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:52)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:38)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerRunnable.run(AbstractDaemonCompiler.java:87)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:36)
	at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:46)
	at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	... 6 more
Caused by: org.gradle.cache.LockTimeoutException: Timeout waiting to lock Zinc 0.3.15 compiler cache (/home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15). It is currently in use by another Gradle instance.
Owner PID: 16501
Our PID: 16540
Owner Operation: 
Our operation: 
Lock file: /home/ubuntu/.gradle/caches/4.3-20170911235713+0000/zinc-0.3.15/zinc-0.3.15.lock
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.lock(DefaultFileLockManager.java:280)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.<init>(DefaultFileLockManager.java:150)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:100)
	at org.gradle.cache.internal.FixedExclusiveModeCrossProcessCacheAccess.open(FixedExclusiveModeCrossProcessCacheAccess.java:67)
	at org.gradle.cache.internal.DefaultCacheAccess.open(DefaultCacheAccess.java:141)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:59)
	... 32 more
```

**Comment by dsilvasc** [comment]
@jjohannes This was on CircleCI with 1 container, with gradle config `javaCompile { options.fork=true }` and `test { forkEvery=25 }`

**Comment by dsilvasc** [comment]
Disabling [incremental compilation](https://docs.gradle.org/4.3/userguide/scala_plugin.html#sec:scala_incremental_compilation) with this config did not help:

```
tasks.withType(ScalaCompile) {
    scalaCompileOptions.with {
        force = true
    }
}
```

**Comment by dsilvasc** [comment]
Is it possible to bypass Zinc entirely and go straight to scalac?  There's no need for incremental compilation during automated (CI) clean builds, tests, or deployments.

**Comment by dsilvasc** [comment]
tracking as a new issue https://github.com/gradle/gradle/issues/3777

**Comment by vincent10111** [comment]
when I use the command: gradle init --type java-application, get the same error and process failed

**Comment by vincent10111** [comment]
When I use the command: gradle init --type java-application, I got the same error, my gradle version is 4.4.1

**Comment by simiyu1** [comment]
I have just encountered the same error... I was performing a gradle sync for two different projects but using the same build version (27.0.2). However, when one project completed its build, the other rebuild worked just fine
